rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -675,9 +660,6 @@ impl TunnelManager {
                     );
                 }
 
-                // In the case that we have a tunnel and they don't we drop our existing one
-                // and agree on the new parameters in this message
-                self.tunnels.remove(&key);
```

Comment:
This removal looks suspicious. Are you sure it should be this way?

Reply:
@mpapierski you remove the key further up right? Here we remove the identity when only a single tunnel may need to be recreated in this special case. This creates orphaned tunnels.